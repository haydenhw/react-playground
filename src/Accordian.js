import React from 'react';

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

class Accordian extends React.Component {
  static defaultProps = {
    sections
  };

  constructor(props) {
    super(props);

    this.state = {
      activeSectionEl: null
    };
  }

  handleClick(index) {
    this.setState({activeSectionEl: index})
  }

  render() {
    const sections = this.props.sections.map((section, index) => {
      return (
        <li key={index}>
          <h1>{section.title}</h1>
          <button onClick={() => this.handleClick(index)}>Expand</button>
          {
            this.state.activeSectionEl === index && <p>{section.content}</p>
          }
        </li>
      )
    });

    return (
      <ul>
        {sections}
      </ul>
    );
  }
}

export default Accordian;
