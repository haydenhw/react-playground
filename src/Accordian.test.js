import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Accordian from './Accordian';

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe('Accordian Component', () => {
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordian />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders only titles given default props', () => {
    const wrapper = shallow(<Accordian />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('expands the first sections when first button is clicked', () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    const firstButton = wrapper.find('button').at(0)
    firstButton.simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
});










