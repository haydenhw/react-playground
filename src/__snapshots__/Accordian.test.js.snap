// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component expands the first sections when first button is clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <h1>
      Section 1
    </h1>
    <button
      onClick={[Function]}
    >
      Expand
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <h1>
      Section 2
    </h1>
    <button
      onClick={[Function]}
    >
      Expand
    </button>
  </li>
  <li
    key="2"
  >
    <h1>
      Section 3
    </h1>
    <button
      onClick={[Function]}
    >
      Expand
    </button>
  </li>
</ul>
`;

exports[`Accordian Component renders only titles given default props 1`] = `
<ul>
  <li
    key="0"
  >
    <h1>
      Section 1
    </h1>
    <button
      onClick={[Function]}
    >
      Expand
    </button>
  </li>
  <li
    key="1"
  >
    <h1>
      Section 2
    </h1>
    <button
      onClick={[Function]}
    >
      Expand
    </button>
  </li>
  <li
    key="2"
  >
    <h1>
      Section 3
    </h1>
    <button
      onClick={[Function]}
    >
      Expand
    </button>
  </li>
</ul>
`;
